api/src/services/items.ts

```diff
@@ -371,7 +371,31 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 	}
 
 	/**
-	 * Update many items by primary key
+	 * Update multiple items in a single transaction
+	 */
+	async updateBatch(data: Partial<Item>[], opts?: MutationOptions): Promise<PrimaryKey[]> {
+		const primaryKeyField = this.schema.collections[this.collection].primary;
+
+		const keys: PrimaryKey[] = [];
+
+		await this.knex.transaction(async (trx) => {
+			const service = new ItemsService(this.collection, {
+				accountability: this.accountability,
+				knex: trx,
+				schema: this.schema,
+			});
+
+			for (const item of data) {
+				if (!item[primaryKeyField]) throw new InvalidPayloadException(`Item in update misses primary key.`);
+				keys.push(await service.updateOne(item[primaryKeyField]!, item, opts));
```

Comment:
I think the `primaryKeyField` within `item` should be omitted when passed to the `updateOne()`. 
Users without update permission for the `primaryKeyField` will result in an error.